src/repository/app-repository.ts

```diff
@@ -0,0 +1,75 @@
+import { Application } from "../model/application";
+import { Company } from "../model/company";
+import { Course } from "../model/course";
+import { ApplicationParams } from "../util/types";
+
+export class ApplicationRepository {
+  async createApplication(applicationParams: ApplicationParams) {
+    try {
+      const { participants, company, course } = applicationParams;
+      if (!company || !course || participants.some((p) => !p)) { 
+        return;
```

Comment:
It is better to return something valuable, in this case it makes sense to be an error? If we just return, the consumer will get a wrong message, may thing that the request passed successfully but in did not.